app/routes/($locale)._index.jsx

```diff
@@ -21,23 +22,59 @@ export async function loader({context}) {
   const featuredCollection = collections.nodes[0];
   const recommendedProducts = storefront.query(RECOMMENDED_PRODUCTS_QUERY);
 
-  return defer({featuredCollection, recommendedProducts});
+  return defer({
+    featuredCollection,
+    recommendedProducts,
+    nostoRecommendations: {},
+  });
 }
 
+export async function clientLoader({serverLoader}) {
+  const recoLoader = () =>
+    new Promise(async (resolve) => {
+      window?.nostojs(async (api) => {
+        const data = await api
+          .defaultSession()
+          .viewFrontPage()
+          .setPlacements(['frontpage-nosto-1', 'frontpage-nosto-3'])
```

Comment:
Here, the placements should be hard-coded. Otherwise, the recommendation rendering may not work properly